src/Mvc/Mvc.Abstractions/src/ModelBinding/ModelMetadata.cs

```diff
@@ -494,6 +475,62 @@ internal IReadOnlyList<ModelMetadata> BoundProperties
         /// </summary>
         public virtual Func<object[], object>? BoundConstructorInvoker => null;
 
+        /// <summary>
+        /// Gets a value that determines if validators can be constructed using metadata exclusively defined on the property.
+        /// </summary>
+        internal virtual bool PropertyHasValidators => false;
+
+        /// <summary>
+        /// Throws if the ModelMetadata is for a record type with validation on properties.
+        /// </summary>
+        internal void ThrowIfRecordTypeHasValidationOnProperties()
+        {
+            CalculateRecordTypeConstructorDetails();
+            if (_recordTypeValidatorsOnPropertiesError != null)
+            {
+                throw _recordTypeValidatorsOnPropertiesError;
+            }
+        }
+
+        [MemberNotNull(nameof(_parameterMapping))]
```

Comment:
Don't believe this attribute is correct. It implies `_parameterMapping` is `null` _before_ the method runs.

Reply:
No, it instructs the compiler that _parameterMapping will always have a value after this method has been called.